Sandbox/Text Adventure/Program.cs

```diff
@@ -0,0 +1,333 @@
+            Console.WriteLine("You quickly jab the spider in its eye and gain an advantage.");
+            randomWait();
+            Console.WriteLine("Fighting!!!!!...");
+            Console.WriteLine("You must hit above a 5 to kill the Spider!");
+            Console.WriteLine("If the spider hits higher than you, you will die! :(");
+            randomWait();
+
+
+        }
+
+        private void msgFightNoStick()
+        {
+            Console.WriteLine("You don't have anything to fight with !!! :(");
+            randomWait();
+            Console.WriteLine("Fighting!!!!!...");
+            Console.WriteLine("You must hit above a 5 to kill the Spider!");
+            Console.WriteLine("If the spider hits higher than you, you will die! :(");
+            randomWait();
+        }
+
+        //random wait generator for text based game
+        private void randomWait()
+        {
+            Random rdm = new Random();
+            int waitTime = rdm.Next(5);
+            for( int x = 0; x < waitTime ; x++ )
+            {
+                Console.WriteLine("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
+
+            }
+
+    }
+
+    //confirms response is a yes within expected response array
+    public  bool confirmYes()
+    {
+        bool confirmation = false; // return false if the answer isn't yes
+        string [] yesArray = new string [] {"yes", "y"};
+        string answer = "";
+        answer = Convert.ToString(Console.ReadLine()).ToLower(); // converts t lowercase
+        if(Array.Exists(yesArray, element => element == answer)) // if the response is in the expected resonse return true
+        {
+
+            confirmation = true;
+
+        }
+
+        return confirmation;
+        
+        }
+
+
+
+
+
+
+     }
+    
+
+    class GameExecution
+    {
+       TextBasedGame game= new TextBasedGame();
+       bool completed = false;
+       bool survivedAlive = false;
+
+       bool replayGame = false;
+
+       private bool runGame()
+       {
+
+           game.gameStart();
+           game.pickUpStick();
+           game.approachSpider();
+           completed = game.completedGame;
+           survivedAlive = completed;
+           return(msgGameEnd(survivedAlive));
+
+       }
+
+       private bool msgGameEnd(bool alive)
+       {
+           if(alive)
+           {
+                Console.WriteLine("You managed to escape the cavern alive! Would you like to play again? [y/n]");
+                replayGame =  game.confirmYes();
+                return replayGame;
+           }
+            else
+            {
+                Console.WriteLine("You have died! Would you like to play again? [y/n]:");
+                replayGame = game.confirmYes();
+                return replayGame;
+
+            }
+
+       }
+
+        public void executeTextBasedGame()
+        {
+            do
+            {
+                runGame();
+            }
+
+            while(replayGame);
```

Comment:
I found a few cases where I was not prompted to play again. ('y' then 'n' for instance). Can you think of a way to rewrite this so you don't accidentally forget to reset state? 

Hint: think about function returns, `out` parameters or new using new types.